ext/src/helpers/nogvl.rs

```diff
@@ -0,0 +1,29 @@
+use std::{ffi::c_void, mem::MaybeUninit, ptr::null_mut};
+
+use rb_sys::rb_thread_call_without_gvl;
+
+unsafe extern "C" fn call_without_gvl<F, R>(arg: *mut c_void) -> *mut c_void
+where
+    F: FnMut() -> R,
+    R: Sized,
+{
+    let arg = arg as *mut (&mut F, &mut MaybeUninit<R>);
+    let (func, result) = unsafe { &mut *arg };
+    result.write(func());
+
+    null_mut()
+}
+
+pub fn nogvl<F, R>(mut func: F) -> R
+where
+    F: FnMut() -> R,
+    R: Sized,
+{
+    let result = MaybeUninit::uninit();
+    let arg_ptr = &(&mut func, &result) as *const _ as *mut c_void;
+
+    unsafe {
+        rb_thread_call_without_gvl(Some(call_without_gvl::<F, R>), arg_ptr, None, null_mut());
```

Comment:
I'm also unsure on signal handling, the doc has warnings about that. I'm not knowledgeable enough in that area to know if it's a concern.

I tried compiling in a tight loop and hitting Control-C and that seemed to work "fine", or at least, no differently than main.

If you want to share your thoughts on that, please do, but I think it's fine?

Reply:
There's no way to interrupt compilation safely, so _during_ compilation interrupts will be ignored which is OK I think